src/snowflake/snowpark/mock/functions.py

```diff
@@ -633,9 +636,60 @@ def mock_endswith(expr1: ColumnEmulator, expr2: ColumnEmulator):
 def mock_row_number(window: TableEmulator, row_idx: int):
     return ColumnEmulator(data=[row_idx + 1], sf_type=ColumnType(LongType(), False))
 
+
 @patch("upper")
 def mock_upper(expr: ColumnEmulator):
     res = expr.apply(lambda x: x.upper())
     res.sf_type = ColumnType(StringType(), expr.sf_type.nullable)
     return res
 
+
+@patch("parse_json")
+def mock_parse_json(expr: ColumnEmulator):
+    if isinstance(expr.sf_type.datatype, StringType):
+        res = expr.apply(lambda x: try_convert(json.loads, False, x))
+    else:
+        res = copy(expr)
```

Comment:
we want a deep copy or shallow copy?
my assumption is that we want a deep copy.

also pandas provides a `copy` on df/series which does a deep copy: https://pandas.pydata.org/docs/reference/api/pandas.Series.copy.html#pandas-series-copy

Reply:
using pandas.dataframe.copy